### PR TITLE
feat: unless, throwError

### PR DESCRIPTION
**Pull request by industriously**
add unless function
add throwError function

this pr is about #215

### CODE REVIEW

File: src/types/ReturnPipeType.ts [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
+  : T["length"] extends 1
+  ? Head<T>
+  : Awaited<T[1]> extends never
+  ? never
+  : PipeLast<Tail<T>>;
+
 type ReturnPipeType<
   T extends any[],
-  R = Awaited<Last<T>>,
+  R = Awaited<PipeLast<T>>,
```
Comment (shine1594) [inline src/types/ReturnPipeType.ts:32]: Could you please write a test case in the `pipe.test.ts` file to verify what improvements this change has made?

Comment (industriously) [reply src/types/ReturnPipeType.ts:32]: 로직 중간에 never 타입이 존재하면, 이후 함수를 고려하지 않고 최종 타입이 never(만약 비동기 작업을 포함한다면 Promise<never>)가 연산되도록 수정했습니다.
그에 대한 케이스도 추가했습니다!

**Review by ppeeou — APPROVED** [review]
lgtm

**Review by shine1594 — COMMENTED** [review]
Personally, I don't think it's easy to infer what a function does just by looking at the name `skip`. In Ramda, there is a function called `unless` that performs the same action, and I think this name is more intuitive. What are your thoughts on changing the function name to [`unless`](https://ramdajs.com/docs/#unless)? @industriously @ppeeou

**Comment by industriously** [comment]
> Personally, I don't think it's easy to infer what a function does just by looking at the name `skip`. In ramda.js, there is a function called `unless` that performs the same action, and I think this name is more intuitive. What are your thoughts on changing the function name to [`unless`](https://ramdajs.com/docs/#unless)? @industriously @ppeeou

i agree!

**Review by shine1594 — APPROVED** [review]
👍

**Comment by ppeeou** [comment]
> > Personally, I don't think it's easy to infer what a function does just by looking at the name `skip`. In ramda.js, there is a function called `unless` that performs the same action, and I think this name is more intuitive. What are your thoughts on changing the function name to [`unless`](https://ramdajs.com/docs/#unless)? @industriously @ppeeou
> 
> i agree!


I agree too
Could you fix the `unless` document as well? @industriously

**Comment by industriously** [comment]
> > > Personally, I don't think it's easy to infer what a function does just by looking at the name `skip`. In ramda.js, there is a function called `unless` that performs the same action, and I think this name is more intuitive. What are your thoughts on changing the function name to [`unless`](https://ramdajs.com/docs/#unless)? @industriously @ppeeou
> > 
> > 
> > i agree!
> 
> I agree too Could you fix the `unless` document as well? @industriously

i commited. please check it.

**Comment by ppeeou** [comment]
Thank you for your contribution 👍

**Pull request merged**